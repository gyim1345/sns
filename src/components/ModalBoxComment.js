import Modal from "react-modal";
import React from "react";
import { css } from "@emotion/core";
import ThreeRoundButton from "./threeRoundButton";
import Edit from "./Edit";
import Remove from "./Remove";
import Reply from "./Reply";

function ModalBoxComment({
  commentAPI,
  indexOfCommentOnThisPosting,
  postings,
  currentUser,
  setCommentAPI,
  addComment
}) {
  const [modalIsOpen, setIsOpen] = React.useState(false);

  function closeModal() {
    setIsOpen(false);
  }

  return (
    <>
      <ThreeRoundButton setIsOpen={setIsOpen} />
      <Modal isOpen={modalIsOpen} onRequestClose={closeModal} css={modalCss}>
        <button onClick={closeModal} css={[marginLeft0]}>
          close
        </button>
        <div>
          <Edit
            posting={commentAPI}
            indexOfCommentOnThisPosting={indexOfCommentOnThisPosting}
            currentUser={currentUser}
            setCommentAPI={setCommentAPI}
          />
          <Remove
            posting={commentAPI}
            currentUser={currentUser}
            indexOfCommentOnThisPosting={indexOfCommentOnThisPosting}
            setCommentAPI={setCommentAPI}
          />

        </div>
      </Modal>
    </>
  );
}

const modalCss = css`
  position: absolute;
  top: 50%;
  left: 50%;
  right: auto;
  bottom: auto;
  border: 1px solid rgb(204, 204, 204);
  background: rgb(255, 255, 255);
  overflow: auto;
  border-radius: 12px;
  outline: none;
  margin-right: -50%;
  transform: translate(-50%, -50%);
  display: flex;
  flex-direction: column-reverse;
`;

const marginLeft0 = css`
  margin-left: 0px;
`;

export default ModalBoxComment;
