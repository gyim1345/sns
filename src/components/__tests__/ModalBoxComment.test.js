import React from 'react';
import { mount } from 'enzyme';
import ModalBoxComment from '../ModalBoxComment';
import { comments, currentUser } from './testjs';

describe('<ModalBoxComment />', () => {
  it('스냅샷 비교', () => {
    const wrapper = mount(
      <ModalBoxComment
        currentUser={currentUser}
        commentAPI={comments}
        indexOfCommentOnThisPosting={0}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
