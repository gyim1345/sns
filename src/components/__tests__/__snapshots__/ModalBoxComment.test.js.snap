// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalBoxComment /> 스냅샷 비교 1`] = `
<ModalBoxComment
  commentAPI={
    Array [
      Object {
        "_id": "5e90119435a2d94f9e08e35a",
        "id": 1,
        "like": Array [
          "noone@gmail.com",
        ],
        "postLId": 1,
        "replyToCommentId": null,
        "title": "comment with postLId 1",
        "userName": "gibong@gmail.com",
      },
    ]
  }
  currentUser="gibong@gmail.com"
  indexOfCommentOnThisPosting={0}
>
  <ThreeRoundButtonSvg
    setIsOpen={[Function]}
  >
    <svg
      aria-label="옵션 더 보기"
      className="_8-yf5 "
      fill="#262626"
      height="16"
      onClick={[Function]}
      style={
        Object {
          "marginRight": "8px",
        }
      }
      viewBox="0 0 48 48"
      width="16"
    >
      <circle
        clipRule="evenodd"
        cx="8"
        cy="24"
        fillRule="evenodd"
        r="4.5"
      />
      <circle
        clipRule="evenodd"
        cx="24"
        cy="24"
        fillRule="evenodd"
        r="4.5"
      />
      <circle
        clipRule="evenodd"
        cx="40"
        cy="24"
        fillRule="evenodd"
        r="4.5"
      />
    </svg>
  </ThreeRoundButtonSvg>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="ModalBoxComment"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIk1vZGFsQm94Q29tbWVudC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUE4Q29CIiwiZmlsZSI6Ik1vZGFsQm94Q29tbWVudC5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBNb2RhbCBmcm9tICdyZWFjdC1tb2RhbCc7XG5pbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IHsgY3NzIH0gZnJvbSAnQGVtb3Rpb24vY29yZSc7XG5cbmltcG9ydCBUaHJlZVJvdW5kQnV0dG9uU3ZnIGZyb20gJy4uL3N2Z0ljb25zL1RocmVlUm91bmRCdXR0b25TdmcnO1xuaW1wb3J0IEVkaXRDb21tZW50IGZyb20gJy4vRWRpdENvbW1lbnQnO1xuaW1wb3J0IFJlbW92ZUNvbW1lbnQgZnJvbSAnLi9SZW1vdmVDb21tZW50JztcblxuZnVuY3Rpb24gTW9kYWxCb3hDb21tZW50KHtcbiAgY29tbWVudEFQSSxcbiAgaW5kZXhPZkNvbW1lbnRPblRoaXNQb3N0aW5nLFxuICBjdXJyZW50VXNlcixcbiAgc2V0Q29tbWVudEFQSVxufSkge1xuICBjb25zdCBbbW9kYWxJc09wZW4sIHNldElzT3Blbl0gPSBSZWFjdC51c2VTdGF0ZShmYWxzZSk7XG5cbiAgZnVuY3Rpb24gY2xvc2VNb2RhbCgpIHtcbiAgICBzZXRJc09wZW4oZmFsc2UpO1xuICB9XG5cbiAgcmV0dXJuIChcbiAgICA8PlxuICAgICAgPFRocmVlUm91bmRCdXR0b25Tdmcgc2V0SXNPcGVuPXtzZXRJc09wZW59IC8+XG4gICAgICA8TW9kYWwgaXNPcGVuPXttb2RhbElzT3Blbn0gb25SZXF1ZXN0Q2xvc2U9e2Nsb3NlTW9kYWx9IGNzcz17bW9kYWxDc3N9PlxuICAgICAgICA8YnV0dG9uIG9uQ2xpY2s9e2Nsb3NlTW9kYWx9IGNzcz17W21hcmdpbkxlZnQwXX0+XG4gICAgICAgICAgY2xvc2VcbiAgICAgICAgPC9idXR0b24+XG4gICAgICAgIDxkaXY+XG4gICAgICAgICAgPEVkaXRDb21tZW50XG4gICAgICAgICAgICBwb3N0aW5nPXtjb21tZW50QVBJfVxuICAgICAgICAgICAgaW5kZXhPZkNvbW1lbnRPblRoaXNQb3N0aW5nPXtpbmRleE9mQ29tbWVudE9uVGhpc1Bvc3Rpbmd9XG4gICAgICAgICAgICBjdXJyZW50VXNlcj17Y3VycmVudFVzZXJ9XG4gICAgICAgICAgICBzZXRDb21tZW50QVBJPXtzZXRDb21tZW50QVBJfVxuICAgICAgICAgIC8+XG4gICAgICAgICAgPFJlbW92ZUNvbW1lbnRcbiAgICAgICAgICAgIHBvc3Rpbmc9e2NvbW1lbnRBUEl9XG4gICAgICAgICAgICBjdXJyZW50VXNlcj17Y3VycmVudFVzZXJ9XG4gICAgICAgICAgICBpbmRleE9mQ29tbWVudE9uVGhpc1Bvc3Rpbmc9e2luZGV4T2ZDb21tZW50T25UaGlzUG9zdGluZ31cbiAgICAgICAgICAgIHNldENvbW1lbnRBUEk9e3NldENvbW1lbnRBUEl9XG4gICAgICAgICAgLz5cbiAgICAgICAgPC9kaXY+XG4gICAgICA8L01vZGFsPlxuICAgIDwvPlxuICApO1xufVxuXG5jb25zdCBtb2RhbENzcyA9IGNzc2BcbiAgcG9zaXRpb246IGFic29sdXRlO1xuICB0b3A6IDUwJTtcbiAgbGVmdDogNTAlO1xuICByaWdodDogYXV0bztcbiAgYm90dG9tOiBhdXRvO1xuICBib3JkZXI6IDFweCBzb2xpZCByZ2IoMjA0LCAyMDQsIDIwNCk7XG4gIGJhY2tncm91bmQ6IHJnYigyNTUsIDI1NSwgMjU1KTtcbiAgb3ZlcmZsb3c6IGF1dG87XG4gIGJvcmRlci1yYWRpdXM6IDEycHg7XG4gIG91dGxpbmU6IG5vbmU7XG4gIG1hcmdpbi1yaWdodDogLTUwJTtcbiAgdHJhbnNmb3JtOiB0cmFuc2xhdGUoLTUwJSwgLTUwJSk7XG4gIGRpc3BsYXk6IGZsZXg7XG4gIGZsZXgtZGlyZWN0aW9uOiBjb2x1bW4tcmV2ZXJzZTtcbmA7XG5cbmNvbnN0IG1hcmdpbkxlZnQwID0gY3NzYFxuICBtYXJnaW4tbGVmdDogMHB4O1xuYDtcblxuZXhwb3J0IGRlZmF1bHQgTW9kYWxCb3hDb21tZW50O1xuIl19 */",
        "name": "102pzl3-modalCss",
        "styles": "position:absolute;top:50%;left:50%;right:auto;bottom:auto;border:1px solid rgb(204,204,204);background:rgb(255,255,255);overflow:auto;border-radius:12px;outline:none;margin-right:-50%;transform:translate(-50%,-50%);display:flex;flex-direction:column-reverse;;label:modalCss;",
        "toString": [Function],
      }
    }
    isOpen={false}
    onRequestClose={[Function]}
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="css-102pzl3-modalCss"
      closeTimeoutMS={0}
      isOpen={false}
      onRequestClose={[Function]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <Portal
        containerInfo={
          <div
            class="ReactModalPortal"
          />
        }
      >
        <ModalPortal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          className="css-102pzl3-modalCss"
          closeTimeoutMS={0}
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={false}
          onRequestClose={[Function]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          role="dialog"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {},
              "overlay": Object {},
            }
          }
        />
      </Portal>
    </Modal>
  </EmotionCssPropInternal>
</ModalBoxComment>
`;
